scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,635 @@
+###
+#
+###############################
+###  Documentation
+###############################
+{
+#
+#      Installer consist of:
+#      1. Setup environment
+#      2. Main constants
+#      3. CLI control constants
+#      4. CLI output functions
+#      5. Program stage functions
+#      6. Main function
+#
+#
+
+# Special things about this script
+#
+# 1) Since this script tries to be fully POSIX compatible, code is influenced by that. There are day-to-day commands and solutions we get used to, but they are not in POSIX standard. So if you try to simplify some operation, please check it with http://pubs.opengroup.org/onlinepubs/9699919799/ .
+#
+# 2) Notice, Warning, Error, ErrorRevert massages have special form to be very readable and get attention.
+# Message body starts from a new line.
+# And has 4 spaces from the left. Always.
+#
+###############################
+#
+#      notice "
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#    "
+#
+###############################
+#
+# This is the best balance of code simplicity and code readability found so far.
+# This produces output in color:
+#
+###############################
+#
+#./install: Notice:
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users independently.
+#
+###############################
+#
+true # so {...} is not empty, shell will not permit otherwise
+}
+###############################
+###  Setup environment
+###############################
+{
+# Set the character collating sequence to be numeric ASCII/C standard.
+# Set the character set to be standard one-byte ASCII.
+# Set default umask; to be non-restrictive and friendly to others.
+#
+LC_COLLATE=C
+LANG=C
+umask 022
+}
+###############################
+###  Main constants
+###############################
+{
+readonly dest="/nix"
+readonly self="$(dirname "$(realpath "$0")")"
+readonly nix="@nix@"
+readonly cacert="@cacert@"
+readonly appname="$0"
+}
+###############################
+###  CLI control constants
+###############################
+{
+if test -t ; then    # File descriptor is associated with a terminal - output colors
+    if tput colors > /dev/null 2>&1 ; then
+        # use tput and terminfo DB
+        readonly red=$(tput setaf 1)
+        readonly green=$(tput setaf 2)
+        readonly yellow=$(tput setaf 3)
+        readonly blue=$(tput setaf 4)
+        readonly bold=$(tput smso)
+        readonly reset=$(tput sgr0) # Reset to default output
+    else
+        # tput is not present on some systems (Alpine Linux), this trick with printf allows to store, not 'codes' - literal symbols
+        readonly red=$(printf '\033[1;31m')
```

Comment:
The motivation was the correct escape codes depending on the used terminal. I am not sure if we still needs this in 2018.

Reply:
* Different distributions, operating systems, different terminals still have differences in color capabilities and color codes.

* There is still a plethora of "$TERM". And terminfo has a big database of different "$TERM" profiles. So `tput` can follow it. Not everything is `xterm-256color`, it can be `linux` or something from BSD. And `linux` does not mean it supports colors. Maybe like some CIs - they use pseudo-terminal, at least we give a chance to blackhole color codes, if CI has right terminfo DB that silences colors. I see a number of small questions covered by this simple code.

If terminal developer supplied DB of color capabilities of current terminal to us, why we omit using them?

Also ASCII codes not follow terminal theme design.
Also `tput` color codes follow design.